src/app/drive/components/DriveExplorer/DriveExplorer.tsx

```diff
@@ -458,31 +466,53 @@ declare module 'react' {
 
 const uploadItems = async (props: DriveExplorerProps, rootList: IRoot[], files: File[]) => {
   const { dispatch, currentFolderId, onDragAndDropEnd } = props;
-  if (files.length) {
-    // files where dragged directly
-    await dispatch(
-      storageThunks.uploadItemsThunkNoCheck({
-        files,
-        parentFolderId: currentFolderId,
-        options: {
-          onSuccess: onDragAndDropEnd,
-        },
-      }),
-    );
-  }
 
-  if (rootList.length) {
-    for (const root of rootList) {
+  let totalFilesToUpload: number;
+
+  totalFilesToUpload = files.length;
+
+  const countTotalItemsToUpload = (rootList) => {
+    rootList.forEach((n) => {
+      totalFilesToUpload += n.childrenFiles.length;
+      if (n.childrenFolders.length >= 1) {
+        countTotalItemsToUpload(n.childrenFolders);
+      }
+    });
+  };
+
+  countTotalItemsToUpload(rootList);
+
+  if (totalFilesToUpload < 1000) {
+    if (files.length) {
+      // files where dragged directly
       await dispatch(
-        storageThunks.uploadFolderThunkNoCheck({
-          root,
-          currentFolderId,
+        storageThunks.uploadItemsThunkNoCheck({
```

Comment:
Why are we changing to `uploadItemsThunkNoCheck` from `uploadFolderThunkNoCheck`. 

I see that `uploadItemsThunkNoCheck` was implemented by larry not so long ago, I think it's a good idea. However this PR only states that we are handling upload size limit, if you are doing any other change please state it in the description of the PR so that it is easier to review. In this case, please add why are we changing to `uploadItemsThunkNoCheck`

Reply:
I have not replaced anything, I have only wrapped the block that already existed in an `if`